### DISCUSSION TITLE
Run Custom Actions with WiX 4.0.5 or 3.14.1?

### DISCUSSION THREAD

**Discussion by MatthiasRInst** [wixtoolset/issues#8185] [Questions]
Hello,

Today I stumbled upon [the bug submitted as #8078 ](https://github.com/wixtoolset/issues/issues/8078), which results in WiX failing to start an impersonated Custom Action because of missing permissions on a temporary folder, showing the error message 
 SFXCA: Failed to create temp directory. Error code 5

I may not have fully understood the preconditions, but from my understanding the only precondition for this error to occur is that an MSI is started without elevation and there is a custom action running with impersonation. So this would affect every immediate-execution CA, doesn't it?

Is this true, or is there a way out without having to upgrade everything to WiX 5 now in a hurry?

Best regards

Matthias

**Comment by barnson** [comment]
You don't have to upgrade everything, just your custom action project.
